packages/editor/src/extensions/code-block/component.tsx

```diff
@@ -27,22 +27,61 @@ import { CodeBlockAttributes } from "./code-block";
 import { ReactNodeViewProps } from "../react/types";
 import { ResponsivePresenter } from "../../components/responsive";
 import { Popup } from "../../toolbar/components/popup";
+import { useIsMobile } from "../../toolbar/stores/toolbar-store";
 import { Button } from "../../components/button";
+import { findParentNodeClosestToPos } from "@tiptap/core";
+import { useTimer } from "../../hooks/use-timer";
 
 export function CodeblockComponent(
   props: ReactNodeViewProps<CodeBlockAttributes>
 ) {
+  const isMobile = useIsMobile();
   const { editor, updateAttributes, node, forwardRef } = props;
-  const { language, indentLength, indentType, caretPosition } = node.attrs;
+  const { language, indentLength, indentType, caretPosition, lines } =
+    node.attrs;
 
   const [isOpen, setIsOpen] = useState(false);
   // const [caretPosition, setCaretPosition] = useState<CaretPosition>();
   const toolbarRef = useRef<HTMLDivElement>(null);
 
+  const { enabled, start } = useTimer(1000);
+
   const languageDefinition = Languages.find(
     (l) => l.filename === language || l.alias?.some((a) => a === language)
   );
 
+  const onClickToCopy = () => {
+    const firstLine = lines.at(0);
+    if (!firstLine) return;
+
+    const { type } = node;
+    const position = editor.state.doc.resolve(firstLine.from);
+    const codeblock = findParentNodeClosestToPos(
+      position,
+      (node) => node.type.name === type.name
+    );
+
+    if (!codeblock) return;
+    const { $from, $to } = editor.state.selection;
+    const finalPosition = codeblock.pos + codeblock.node.nodeSize - 1;
+    const isNotSelecting = $from.pos !== codeblock.pos + 1 || $to.pos !== finalPosition
+
+    if (isNotSelecting) {
+      editor.commands.setTextSelection({
+        from: codeblock.pos + 1,
+        to: finalPosition
+      });
+    }
+    editor.commands.focus();
+    editor.commands.copy(codeblock.node.textContent);
+    start();
+  };
```

Comment:
This all is unnecessary. Inline the onClick handler code & just call `writeText(node.textContent)`.